src/data/int/order.lean

```diff
@@ -89,4 +89,17 @@ begin
   exact dif_pos ⟨h1, h2⟩,
 end
 
+/-- Induction principle on `ℤ` for all integers above a given integer `m`. -/
+@[elab_as_eliminator] lemma le_induction {P : ℤ → Prop} {m : ℤ}
```

Comment:
The current proof works for `Prop` only, so I prefer to not do this in this PR.

Reply:
Mathlib's #data.nat.basic has `nat.le_induction` that works for `Prop`, and `nat.le_rec_on` that works in general. Why not the same for `int` ?

Reply:
(I am not sure I understand the issue, but using `obtain` or other `dcases` eliminators seems to be restricted for `Prop`-valued functions. I hoped eliminating a `exists.unique` would help, but it does not seem to do.

Reply:
(Now, I guess that there is a computability issue, but it should not be serious in the case of `int` where equality is decidable.)

Reply:
It is definitely possible to do define `int.le_rec[_on]` as well, but it takes some care if you want it to compute in a good way. For `nat` this is easier, since the inequality is defined directly as an inductive type (see `nat.less_than_or_equal`).